### ISSUE TITLE
Icons of wine applications aren't displayed

### ISSUE THREAD

**Issue by CreamyCookie** [robint99/mate-dock-applet#152]
The icons of wine applications aren't rendered. Instead the standard wine icon is shown. There also a cog icon, that cannot be clicked:

![wine screenshot](https://user-images.githubusercontent.com/3063858/41823703-fb976dde-7804-11e8-885a-63321fd20b01.png)

 The desktop file (created by PlayOnLinux) does show an icon. (It's an image file without an extension.)

**Comment by robint99** [comment]
Which distro and applet version are you using? I don't have many games that I can run under PlayOnLinux, but I tried Pharaoh and Cleopatra and that at least did show the correct icon.
 
![pharoah](https://user-images.githubusercontent.com/10033637/42308961-32c8d72a-802f-11e8-93cc-5ca331aa6e02.jpg)

This was on Ubuntu Mate 18.04 with V0.86 of the applet.

**Comment by CreamyCookie** [comment]
I'm also using Ubuntu Mate 18.04, but my applet version was V0.85. I've now manually installed from master (which looks great with the high res icons). I've tried with the Battle.net launcher and it just displays a set of cogs, while the Ubuntu Mate alt-tab-switcher displays the correct icon:


![screenshot](https://user-images.githubusercontent.com/3063858/42314100-92af4902-8044-11e8-9ad5-ffe64c604714.png)

**Comment by robint99** [comment]
I've just committed a fix for this. Wine apps now show the Wine icon defined by the icon theme e.g.

Ambient Mate: 
![screenshot at 2018-07-17 08-15-32](https://user-images.githubusercontent.com/10033637/42802350-931064b2-899a-11e8-9c23-7b5bf069e614.png)

Adapta:
![screenshot at 2018-07-17 08-13-26](https://user-images.githubusercontent.com/10033637/42802358-9b60ec04-899a-11e8-8704-91301803f282.png)

As for the invisible windows issues, there seems to be a long standing issue with PlayOnLinux and Freetype (e.g. see [here](https://bbs.archlinux.org/viewtopic.php?id=224534). Basically, because fonts can't be rendered you end up with windows only a few pixels in height and width and these are difficult to see. I ran into exactly this problem when installing Pharoah and when trying to install BattleNet. Hopefully the PlayOnLinux devs will sort this out soon...

I'll close this issue as the icons are now being displayed properly.

**Comment by CreamyCookie** [comment]
@robint99 Thanks, I've just tried it. 

Wouldn't it also be possible to show the right icon like the Mate Alt-Tab switcher does, as shown here:

![screenshot](https://user-images.githubusercontent.com/3063858/42314100-92af4902-8044-11e8-9ad5-ffe64c604714.png)

**Comment by robint99** [comment]
It may be that with the new code the Battle Net icon will be displayed correctly. Unfortunately, because of the PlayOnLinux font issue I mentioned above, I haven't been to install it and do any testing...

If you want to try it yourself I've written a Python script to install the latest development version of the applet do this for you, and you can find it [here](https://github.com/robint99/misc) in one of my other repositories.

Cheers.

**Comment by CreamyCookie** [comment]
Oh, sorry I wasn't clear. I had compiled from master and it's not showing the battle net one, just the default wine one.

**Comment by robint99** [comment]
After several attempts I was able to get the Battle Net installer to run, and the icon was indeed wrong. I've updated the code again and now the installer does have the correct icon.

![screenshot at 2018-07-19 08-36-36](https://user-images.githubusercontent.com/10033637/42928795-2d9d6ff2-8b30-11e8-9ebf-25acd66cab4e.png)

The installer errors out but so I can't test Battle Net itself, but the changes I've made should mean that Battle Net and any other Wine app that sets an icon for itself should show that icon in the dock.

I'll reopen this issue until you have a chance to test if for yourself ...

Cheers
